File: leetcode/pascalTriangle2.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        tmp = [1]
        
        for i in range(0, rowIndex + 1):
            prev = tmp
            tmp = [0] * (i + 1)
            for j in range(0, i + 1):
                if(j == 0):
                    tmp[j] = prev[j]
                elif(j == i):
                    tmp[j] = prev[j-1]
                else:
                    tmp[j] = prev[j-1] + prev[j]
                
            if(i == rowIndex): return tmp
        
        return tmp
            

File: leetcode/pascalTriangle2.java
class Solution {
  public List<Integer> getRow(int rowIndex) {
      
      int[] prev;
      int[] tmp = new int[1];
      tmp[0] = 1;
      
      List<Integer> ans = new ArrayList<>();
      
      for(int i = 0; i <= rowIndex; i++) {
          prev = tmp;
          tmp = new int[i+1];
          for(int j=0; j <= i; j++) {
              if(j == 0) tmp[j] = prev[j];
              else if(j == i) tmp[j] = prev[j-1];
              else tmp[j] = prev[j] + prev[j-1];
          }
          
          if(i == rowIndex) {
              for(int num:tmp) ans.add(num);
              break;
          }
      }
      
      return ans;
  }
}